src/data/set/intervals/interval.lean

```diff
@@ -0,0 +1,137 @@
+/-
+Copyright (c) 2020 Zhouhang Zhou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Zhouhang Zhou
+-/
+
+import data.set.intervals
```

Comment:
It feels weird to have a file `data.set.intervals`, and another `data.set.intervals.interval`. Maybe move the new file to `data.set.unordered_intervals`, or have a subdirectory containing both files?